src/Plugin/facets/query_type/LocalGovDirectoriesQueryType.php

```diff
@@ -63,11 +64,48 @@ public function execute() {
    * {@inheritdoc}
    */
   public function build() {
+    $results = $this->results;
     $query_operator = $this->facet->getQueryOperator();
+    $field_identifier = $this->facet->getFieldIdentifier();
 
-    if (!empty($this->results)) {
+    // Get passed in facets.
+    $temp_query = $this->query->getOriginalQuery();
+    $temp_query->preExecute();
+    $conditions = &$temp_query->getConditionGroup()->getConditions();
+
+    // Find all the facet condition groups.
+    $facet_conditions = [];
+    foreach ($conditions as $key => $condition) {
+      if ($condition instanceof ConditionGroupInterface) {
+        $tags = $condition->getTags();
+        foreach($tags as $tag) {
+          if (strpos($tag, 'facet:' . $field_identifier . '.') === 0) {
+            $facet_conditions[$key] = $tag;
+          }
+        }
+      }
+    }
+
+    // Run query elimnating each facet group and return the resulting facets.
+    $results = $this->results;
```

Comment:
Duplicates https://github.com/localgovdrupal/localgov_directories/pull/269/files#diff-0872c6a42a990750a6b255b18dd8466e3dc6af43670c3755fae3c1e155fca46dR67 and we've not used it yet?